js/DropdownButtonComponent.js

```diff
@@ -0,0 +1,219 @@
+    }
+
+    /**
+     * Handle toggle
+     * @param {Event} event
+     */
+    toggle (event) {
+        const $button = $(event.target).closest(Selectors.DATA_TOGGLE),
+            $parent = $button.parent(),
+            dropdownOpen = $parent.hasClass(Selectors.OPEN);
+
+        if ($button.is(Selectors.DISABLED)) {
+            return;
+        }
+
+        this.closeAllDropdowns();
+
+        if (!dropdownOpen) {
+            if (event.isDefaultPrevented()) {
+                return;
+            }
+
+            $parent.toggleClass(Selectors.OPEN);
+            $button.attr('aria-expanded', 'true');
+        }
+    }
+
+    /**
+     * Close all dropdowns
+     */
+    closeAllDropdowns () {
+        const dropdownButtons = this.$html.find(Selectors.DATA_TOGGLE);
+
+        dropdownButtons.each((index, button) => {
+            let $button = $(button),
+                $parent = $button.parent();
+
+            if (!$parent.hasClass(Selectors.OPEN)) {
+                return;
+            }
+
+            $parent
+                .removeClass(Selectors.OPEN)
+                .trigger('hidden.bs.dropdown');
+
+            $button.attr('aria-expanded', 'false');
+        });
+    }
+
+    /**
+     * Close all dropdowns on body click
+     * @param {Event} event
+     */
+    handleBodyClick (event) {
+        const $clickedElement = $(event.target);
+
+        if ($clickedElement.attr('data-toggle') === 'dropdown' || $clickedElement.closest(Selectors.BTN_GROUP).hasClass(Selectors.OPEN)) {
+            return;
+        }
+        this.closeAllDropdowns();
+    }
+
+    /**
+     * Close all dropdowns on body click
+     * @param {Event} event
+     */
+    handleEscKeydown (event) {
+        const $elementWithFocus = $(':focus');
+
+        if (event.keyCode === Keycodes.ESC) {
+            if ($elementWithFocus.closest(Selectors.BTN_GROUP_DROP)) {
+                $elementWithFocus.closest(Selectors.BTN_GROUP_DROP).find(Selectors.DATA_TOGGLE).focus();
+            }
+
+            this.closeAllDropdowns();
+        }
+    }
+
+    /**
+     * Open dropdown menu on arrow up or down, depending on dropdown or dropup
+     * @param {Event} event
+     */
+    handleDropButtonKeydown (event) {
+        const $keydownElement = $(event.target),
+            $parentButtonGroup = $keydownElement.closest(Selectors.BTN_GROUP),
+            direction = $parentButtonGroup.hasClass(Selectors.DROPUP) ? 'up' : 'down',
+            $dropdownItems = $parentButtonGroup.find(Selectors.VISIBLE_ITEMS);
+
+        if ($keydownElement.is(Selectors.DISABLED)) {
+            return;
+        }
+
+        if (event.keyCode === Keycodes.DOWN && direction === 'down') {
+            event.preventDefault();
+
+            if (!$dropdownItems.length) {
+                return;
+            }
+
+            if (!$parentButtonGroup.hasClass(Selectors.OPEN)) {
+                $keydownElement.click();
```

Comment:
`.click()` shorthand method deprecated in jQuery 3, could we change this to a `.trigger('click')` in anticipation?